apps/platform/src/providers/MessageTriggerService.ts

```diff
@@ -70,7 +74,31 @@ export async function loadSendJob<T extends TemplateType>({ campaign_id, user_id
         user_step_id,
     }
 
-    return { campaign, template: template.map() as T, user, project, event, context }
+    // provide data captured from linked journey steps
+    let journey: Record<string, unknown> = {}
+    if (user_step_id) {
+        let step = await JourneyUserStep.find(user_step_id)
+        if (step) {
+            if (step.entrance_id) {
+                step = (await JourneyUserStep.find(step.entrance_id))!
+            }
```

Comment:
If we had a re-use method for finding the entrance we could use it in places like this.